src/test/java/com/google/research/bleth/simulator/ObserverTest.java

```diff
@@ -0,0 +1,930 @@
+package com.google.research.bleth.simulator;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.runner.RunWith;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.Mock;
+
+import java.util.List;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ObserverTest {
+    private static final Location zeroOnZeroCoordinate = new Location(0, 0);
+    private static final Location oneOnOneCoordinate = new Location(1, 1);
+    private static final Location zeroOnOneCoordinate = new Location(0, 1);
+    private static final Location oneOnZeroCoordinate = new Location(1, 0);
+    private static final BeaconFactory beaconFactory = new BeaconFactory();
+    private static final ObserverFactory observerFactory = new ObserverFactory();
+
+    @Mock
+    private Simulation simulation;
+
+    /**
+     * A fake implementation for IResolver that allows to test the arguments it receives when an observer calls receiveInformation.
+     */
+    private class FakeResolver implements IResolver {
+        private Location observerLocation;
+        private List<Transmission> transmissions;
+
+        @Override
+        public void receiveInformation(Location observerLocation, List<Transmission> transmissions) {
+            this.observerLocation = observerLocation;
+            this.transmissions = transmissions;
+        }
+
+        public Location getObserverLocation() {
+            return observerLocation;
+        }
+
+        public List<Transmission> getTransmissions() {
+            return transmissions;
+        }
+
+        @Override
+        public Board getBoard() {
+            return null;
+        }
+    }
+
+    private IResolver resolver = new FakeResolver();
+
+    private Observer createStaticObserverOnLocation(Location initial_location) {
```

Comment:
Prefer to place helper methods below the tests: they make more sense once someone already knows what they're used for. (Also, the most important part of a test class are the tests - you want a reader to get to them early.)